src/argilla/training/setfit.py

```diff
@@ -0,0 +1,157 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import json
+from typing import List, Union
+
+from datasets import DatasetDict
+
+from argilla.training.utils import filter_allowed_args, get_default_args
+
+
+class ArgillaSetFitTrainer(object):
+    # @require_version("setfit", "0.6")
+    def __init__(self, dataset, device, record_class, multi_label: bool = False, seed: int = None):
+        from setfit import SetFitModel, SetFitTrainer
+
+        if seed is None:
+            seed = 42
+        self._seed = seed
+        self._record_class = record_class
+        self._multi_label = multi_label
+
+        if isinstance(dataset, DatasetDict):
+            self._train_dataset = dataset["train"]
+            self._eval_dataset = dataset["test"]
+        else:
+            self._train_dataset = dataset
+            self._eval_dataset = None
+
+        if multi_label:
+            self.multi_target_strategy = "one-vs-rest"
+            self._column_mapping = {"text": "text", "binarized_label": "label"}
+        else:
+            self.multi_target_strategy = None
+            self._column_mapping = {"text": "text", "label": "label"}
+
+        if self._multi_label:
+            self._id2label = dict(enumerate(self._train_dataset.features["label"][0].names))
+        else:
+            self._id2label = dict(enumerate(self._train_dataset.features["label"].names))
+        self._label2id = {v: k for k, v in self._id2label.items()}
+
+        self.setfit_model_kwargs = get_default_args(SetFitModel.from_pretrained)
+        self.setfit_model_kwargs["pretrained_model_name_or_path"] = "all-MiniLM-L6-v2"
```

Comment:
The choice here has notable impact on performance. Do you want people to be able to update this only via `update_config`? It's fine if so, I'm just wondering.

Reply:
are you talking about the model? Assuming that we focus on people that want to get an easy approach I wanted to use a quick-and-dirty baseline. 
- What model would you suggest? 
- Perhaps the choice of model can be added as a specific parameter to the `BaseTrainer` to make it more flexible, however, I prefer having most config-related things go via the update config function.

Reply:
We tend to use `"sentence-transformers/paraphrase-mpnet-base-v2"` for our examples. It's a bit larger than "all-MiniLM-L6-v2". And I'm fine with just using the `update_config` method to update the params.